### DISCUSSION TITLE
Existing simulator for G02/G03 G-Code

### DISCUSSION THREAD

**Discussion by 5axes** [FormerLurker/ArcWelderLib#75] [Q&A] [answered]
Hello,

Is it possible to have a list of solution available to simulate G02/G03 G-Code ?  
I'm using https://ncviewer.com/  but it's not the best solution to simulate G-Code for additive technology.
G-Code Q'n'dirty toolpath simulator can be also used but it's very difficult to zoom on the toolpath https://nraynaud.github.io/webgcode/

regards,

**Comment by FormerLurker** [comment, marked as answer]
Sure, but none do everything you'll want.  

Ncviewer is nice to zoom in on a particular piece of gcode while showing the endpoints of the toolpaths.  It gets confused by some gcodes, and other stuff like embedded thumbnails.  I frequently will only paste in certain parts of a gcode file that I want to scrutinize, since having all of the code for a large object makes it difficult to interpret.

Simplify3D is very good, and is the most accurate visualizer, IMHO.  It lets you trim off gcode with a slider, but does not show the code.  It has no layer by layer view.

OctoPrint's viewer does a great job showing the toolpaths on a layer-by-layer basis.

The PrettyGcode plugin for Octoprint renders the toolpaths in a similar way to how Marlin will interpolate the gcode, so it's closer to what will actually print, and it will render the entire model all at once, or show you progress for the current print.

I also like to use the 'ArcStraightener' binary (included in this repo) to convert G2/G3 to G1/G2 for visualization.  I'm in the process of updating this to output the toolpaths for many different kinds and versions of firmware, including Marlin 1, Marlin 2, Prusa's Firmware for the Mk2/Mk3, smoothieware, repetier, and maybe more.  This is very useful for debugging and the like, and the resulting gcode can be visualized pretty much anywhere. 

I hope that helps!

**Comment by 5axes** [comment]
The G-Code Viewer of SuperSlicer support now G2/G3 GCode. But if the visualisation is correct we don't have any indication if it's a G02 or G03 G-Code. 
https://github.com/supermerill/SuperSlicer/releases/tag/2.3.57.0

**Comment by FormerLurker** [comment]
@5axes, Excellent!  I'm going to have to try it out, thanks for letting me know.
